Src/AutoFixture/EmailAddressLocalPart.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Represents the local part of the email address, defined as everything up to, but not including the @ sign.  
+    /// The requirements enforced on the local part are defined in RFC 3696.  The regular expression for validating 
+    /// the local part borrowed from Phil Haack at 
+    /// http://haacked.com/archive/2007/08/21/i-knew-how-to-validate-an-email-address-until-i.aspx/
```

Comment:
@ploeh When I move `EmailAddressLocalPart` to be a public inner class of `MailAddressGenerator`, the code analysis in the BuildRelease script throws an error as it only allows non-public inner classes.  Do you feel strongly enough that we should nest `EmailAddressLocalPart` that we should suppress the rule in this case or do you take that as evidence we should keep it in the outer scope despite the coupling?  For whatever it's worth, I'll note that the MSDN article on that warning advises to not suppress it - https://msdn.microsoft.com/en-us/library/ms182162.aspx